### DISCUSSION TITLE
Difference between window.serwist and @serwist/window

### DISCUSSION THREAD

**Discussion by HugeLetters** [serwist/serwist#130] [Q&A] [answered]
Is there any difference between using `window.serwist` and using `@serwist/window` [module](https://serwist.pages.dev/docs/window)? 

The API seems identical to me.
Is the benefit that the module will have a more straight-forward TS support? Does add anything extra to the bundle?

Or is the difference here that I have `window.serwist` available because that was already handled by nextjs module and `@serwist/window` is intended to be used when you're handling the build on your own?

**Comment by DuCanhGH** [comment, marked as answer]
@HugeLetters if you use `@serwist/next` then `@serwist/window` is not needed. `window.serwist` is pretty much that but initialized for you.

To be honest though, I do regret having not gotten rid of `@serwist/next`'s `window.serwist` in 9.0.0. Feels like we are introducing some unjustified and weird complexity to the codebase and the API surface.
